indra/databases/ndex_client.py

```diff
@@ -67,3 +70,129 @@ def send_request(ndex_service_url, params, is_json=True, use_get=False):
+        'user': NDEx user name
+        'password': NDEx password
+
+    Returns
+    -------
+    network_id :  str
+        The UUID of the NDEx network that was created by uploading
+        the assembled CX model.
+    """
+    nd = ndex.client.Ndex('http://public.ndexbio.org',
+                          username=ndex_cred.get('user'),
+                          password=ndex_cred.get('password'))
+    cx_stream = io.BytesIO(cx_str.encode('utf-8'))
+    try:
+        logger.info('Uploading network to NDEx.')
+        network_uri = nd.save_cx_stream_as_new_network(cx_stream)
+    except Exception as e:
+        logger.error('Could not upload network to NDEx.')
+        logger.error(e)
+        return
+
+    network_id = network_uri.rsplit('/')[-1]
+    logger.info('The UUID for the uploaded network is: %s' % network_id)
+    logger.info('View at: http://ndexbio.org/#/network/%s' % network_id)
+    return network_id
+
+
+def update_network(cx_str, network_id, ndex_cred):
+    """Update an existing CX network on NDEx with new CX content.
+
+    Parameters
+    ----------
+    cx_str : str
+        String containing the CX content.
+    network_id : str
+        UUID of the network on NDEx.
+    ndex_cred : dict
+        A dictionary with the following entries:
+        'user': NDEx user name
+        'password': NDEx password
+    """
+    server = 'http://public.ndexbio.org'
+    username = ndex_cred.get('user')
+    password = ndex_cred.get('password')
+    nd = ndex.client.Ndex(server, username, password)
+
+    try:
+        logger.info('Getting network summary...')
+        summary = nd.get_network_summary(network_id)
+    except Exception as e:
+        logger.error('Could not get NDEx network summary.')
+        logger.error(e)
+        return
+
+    # Update network content
+    try:
+        logger.info('Updating network...')
+        cx_stream = io.BytesIO(cx_str.encode('utf-8'))
+        nd.update_cx_network(cx_stream, network_id)
+    except Exception as e:
+        logger.error('Could not update NDEx network.')
+        logger.error(e)
+        return
+
+    # Update network profile
+    ver_str = summary.get('version')
+    new_ver = _increment_ndex_ver(ver_str)
+    profile = {'name': summary.get('name'),
+               'description': summary.get('description'),
+               'version': new_ver,
+               }
+    logger.info('Updating NDEx network (%s) profile to %s' %
+                (network_id, profile))
+    profile_retries = 5
+    for i in range(profile_retries):
+        try:
+            time.sleep(5)
+            nd.update_network_profile(network_id, profile)
+            break
+        except Exception as e:
+            logger.error('Could not update NDEx network profile.')
+            logger.error(e)
+
+    # Update network style
+    import ndex.beta.toolbox as toolbox
+    template_uuid = "ea4ea3b7-6903-11e7-961c-0ac135e8bacf"
+
+    d_edge_types = ["Activation", "Inhibition",
```

Comment:
We can actually remove this d_edge_types thing, it's left over from something else.